src/Workspaces/Core/Portable/Workspace/Solution/AssetHint.cs

```diff
@@ -13,19 +13,41 @@ namespace Microsoft.CodeAnalysis.Serialization;
 [DataContract]
 internal readonly struct AssetHint
 {
-    public static readonly AssetHint None = default;
+    /// <summary>
+    /// Instance that will only look up solution-level data when searching for checksums.
+    /// </summary>
+    public static AssetHint SolutionOnly = default;
+
+    /// <summary>
+    /// Instance that will only look up solution-level, as well as the top level nodes for projects when searching for
+    /// checksums.  It will not descend into projects.
+    /// </summary>
+    public static AssetHint SolutionAndTopLevelProjectsOnly = new(projectId: null, documentId: null, topLevelProjects: true, isFullLookup_ForTestingPurposesOnly: false);
```

Comment:
this is used in one place in bulk synchronization.  i'm skeptical it is needed (vs just requesting each project one at a time).  but i didn't want to change that logic in this pr.  a followup will change and measure if there are any issues.